src/classes/DataFactory.cls

```diff
@@ -18,11 +45,17 @@ public abstract class DataFactory {
     
     abstract SObject build();
     
-	SObject create() {
+	public SObject create() {
         SObject item = build();
+        countInstances++;
+        prePopulate(item);
         insert item;
```

Comment:
This DML statement is not surrounded by any try-catch block. Please add a try-catch block to ensure exceptions are handled properly.